src/app/query/bitcoin/ordinals/use-inscriptions.query.ts

```diff
@@ -48,18 +54,26 @@ export function useInscriptionsInfiniteQuery() {
   const network = useCurrentNetwork();
   const keychain = useTaprootCurrentAccountPrivateKeychain();
   const currentAccountIndex = useCurrentAccountIndex();
+  // we need to save the current account index to reset the account data when the account changes
+  const savedCurrentAccountIndex = useRef(currentAccountIndex);
 
   // TO-DO remove code for testing purposes before merge
   const searchParams = new URLSearchParams(document.location.search);
   const testInscriptionAddress = searchParams.get('testAddress');
-  const currentNumberOfAddressesWithoutOrdinals = useRef(0);
-
-  const fromIndex = useRef(0);
-  const addressesMap = useRef(
-    getAddressesData(fromIndex.current, fromIndex.current + addressesSimultaneousFetchLimit)
-  );
-  const queryParams = useRef(formQueryParams(addressesMap.current));
 
+  const defaultAccountData = {
+    fromIndex: 0,
+    addressesWithoutOrdinalsNum: 0,
+    addressesMap: getAddressesData(0, addressesSimultaneousFetchLimit),
+  };
+  const currentAccData = useRef<AccountData>(defaultAccountData);
+  const queryParams = useRef(formQueryParams(currentAccData.current.addressesMap));
+
+  if (savedCurrentAccountIndex.current !== currentAccountIndex) {
+    currentAccData.current = defaultAccountData;
+    queryParams.current = formQueryParams(currentAccData.current.addressesMap);
+    savedCurrentAccountIndex.current = currentAccountIndex;
+  }
```

Comment:
This seems like functionality that belongs in an effect

```ts
useEffect(() => {
  currentAccData.current = defaultAccountData;
  queryParams.current = formQueryParams(currentAccData.current.addressesMap);
  savedCurrentAccountIndex.current = currentAccountIndex;
}, [currentAccountIndex]);
```
?

Reply:
exactly,  with `useEffect` here, we don't need to save current acc index, thanks

we need to clear data on acc change as otherwise, e.g `addressesWithoutOrdinalsNum` and will remain same and query function in while loop won't work